apps/api/src/persistence/repositories/arango-data-exporter.spec.ts

```diff
@@ -0,0 +1,177 @@
+
+const testDocs = [
+    {
+        _key: '123',
+        name: 'bobby',
+    },
+    {
+        _key: '124',
+        name: 'billy',
+    },
+    {
+        _key: '125',
+        // ensure unicode chars come through
+        name: 'ŵŝẑʔɨ’',
+    },
+];
+
+const testEdges: HasArangoDocumentDirectionAttributes<
+    ArangoDocumentForAggregateRoot<HasAggregateId>
+>[] = [
+    {
+        _key: '333',
+        _to: 'docs/123',
+        _from: 'docs/124',
+    },
+    {
+        _key: '334',
+        _to: 'docs/123',
+        _from: 'docs/123',
+    },
+];
+
+const dir = `__cli-command-test-files__`;
+
+const filename = 'data-dump__dump-file__.data.json';
+
+const fullpath = path.join(dir, filename);
+
+describe(`ArangoDataExporter`, () => {
+    let app: INestApplication;
+
+    let arangoDataExporter: ArangoDataExporter;
+
+    let arangoDatabaseProvider: ArangoDatabaseProvider;
+
+    const testEdgeCollectionName = 'edges';
+
+    const testDocumentCollectionName = 'docs';
+
+    const knownEdgeCollections = [...Object.values(ArangoEdgeCollectionId), testEdgeCollectionName];
+
+    beforeAll(async () => {
+        const testModule = await Test.createTestingModule({
+            imports: [ConfigModule, PersistenceModule.forRootAsync()],
+        })
+            .overrideProvider(ConfigService)
+            .useValue(
+                buildMockConfigService(
+                    {
+                        ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+                    },
+                    buildConfigFilePath(Environment.test)
+                )
+            )
+            .compile();
+
+        app = testModule.createNestApplication();
+
+        await app.init();
+
+        arangoDatabaseProvider = app.get(ArangoDatabaseProvider);
+
+        arangoDataExporter = app.get(ArangoDataExporter);
+    });
+
+    beforeEach(async () => {
+        await app
+            .get(ArangoConnectionProvider)
+            .createCollectionIfNotExists(testDocumentCollectionName);
+
+        await arangoDatabaseProvider.getDatabaseForCollection(testDocumentCollectionName).clear();
+
+        await arangoDatabaseProvider
+            .getDatabaseForCollection(testDocumentCollectionName)
+            .createMany(testDocs);
+
+        await app.get(ArangoConnectionProvider).createCollectionIfNotExists(testEdgeCollectionName);
+
+        await arangoDatabaseProvider.getDatabaseForCollection(testEdgeCollectionName).clear();
+
+        await arangoDatabaseProvider
+            .getDatabaseForCollection(testEdgeCollectionName)
+            .createMany(testEdges);
+
+        if (existsSync(fullpath)) {
+            unlinkSync(fullpath);
+        }
+
+        const testRepositoryProvider = new TestRepositoryProvider(
+            arangoDatabaseProvider,
+            app.get(CoscradEventFactory),
+            app.get(DynamicDataTypeFinderService)
+        );
+
+        await testRepositoryProvider.testSetup();
+
+        // TODO add some views
```

Comment:
Should we do this? Note that it's not a big risk as you can always rehydrate the views.